src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
+
+      if (this[i] === undefined || this[i] === null) {
+        item = '';
+      }
```

Comment:
You do not need to declare the value once again and overwrite it if null or undefined comes to you.
You can use `!==` in `if` condition and if value exist add it to result string.

Reply:
This code redunant here. Just check what you get when you use built-in `join` method, when you nave `null` or `undefined` in your array.